ApplicationLibCode/ProjectDataModel/Faults/RimFaultReactivationDataAccessorStressEclipse.cpp

```diff
@@ -0,0 +1,305 @@
+#include "RigWellPath.h"
+
+#include "RimEclipseCase.h"
+#include "RimFaultReactivationDataAccessorStress.h"
+#include "RimFaultReactivationDataAccessorWellLogExtraction.h"
+#include "RimFaultReactivationEnums.h"
+
+#include "cvfObject.h"
+#include "cvfVector3.h"
+
+#include <cmath>
+#include <limits>
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimFaultReactivationDataAccessorStressEclipse::RimFaultReactivationDataAccessorStressEclipse(
+    RimEclipseCase*                                            eclipseCase,
+    RimFaultReactivation::Property                             property,
+    double                                                     gradient,
+    double                                                     seabedDepth,
+    double                                                     lateralStressComponentX,
+    double                                                     lateralStressComponentY,
+    const std::map<RimFaultReactivation::ElementSets, double>& densities )
+    : RimFaultReactivationDataAccessorStress( property, gradient, seabedDepth )
+    , m_eclipseCase( eclipseCase )
+    , m_caseData( nullptr )
+    , m_mainGrid( nullptr )
+    , m_lateralStressComponentX( lateralStressComponentX )
+    , m_lateralStressComponentY( lateralStressComponentY )
+    , m_densities( densities )
+{
+    if ( m_eclipseCase )
+    {
+        m_caseData = m_eclipseCase->eclipseCaseData();
+        m_mainGrid = m_eclipseCase->mainGrid();
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimFaultReactivationDataAccessorStressEclipse::~RimFaultReactivationDataAccessorStressEclipse()
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimFaultReactivationDataAccessorStressEclipse::updateResultAccessor()
+{
+    if ( !m_caseData ) return;
+
+    RigEclipseResultAddress resVarAddress( RiaDefines::ResultCatType::DYNAMIC_NATIVE, "PRESSURE" );
+    m_eclipseCase->results( RiaDefines::PorosityModelType::MATRIX_MODEL )->ensureKnownResultLoaded( resVarAddress );
+
+    m_resultAccessor =
+        RigResultAccessorFactory::createFromResultAddress( m_caseData, 0, RiaDefines::PorosityModelType::MATRIX_MODEL, m_timeStep, resVarAddress );
+
+    auto [wellPaths, extractors] =
+        RimFaultReactivationDataAccessorWellLogExtraction::createEclipseWellPathExtractors( *m_model, *m_caseData, m_seabedDepth );
+    m_wellPaths  = wellPaths;
+    m_extractors = extractors;
+
+    // TODO: get from model
+    double waterDensity = 1030.0;
```

Comment:
Should use the value from the RimFaultReactivationModel object?

Reply:
Fixed.